ts/components/RemindEmailValidationOverlay.tsx

```diff
@@ -0,0 +1,158 @@
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+type State = {
+  dispatched: boolean;
+};
+
+class RemindEmailValidationOverlay extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      dispatched: false
+    };
+  }
+
+  private handleBackPress = () => {
+    this.props.navigateBack();
+    return true;
+  };
+
+  public componentDidMount() {
+    BackHandler.addEventListener("hardwareBackPress", this.handleBackPress);
+  }
+
+  public componentWillUnmount() {
+    BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
+  }
+
+  public componentDidUpdate(prevprops: Props, prevstate: State) {
+    const { isValidEmail } = this.props;
+    const { dispatched } = this.state;
+    if (
+      !prevprops.isValidEmail &&
+      !isValidEmail &&
+      prevstate.dispatched &&
+      dispatched
+    ) {
+      this.setState(
+        {
+          dispatched: false
+        },
+        this.props.navigateBack
+      );
+    }
+  }
+
+  public render() {
+    const { optionProfile } = this.props;
+    const profileEmail = optionProfile
+      .map(_ => untag(_.spid_email))
+      .getOrElse("");
+    return (
+      <TopScreenComponent
+        customRightIcon={{
+          iconName: "io-close",
+          onPress: this.props.navigateBack
+        }}
+      >
+        <Content>
+          <Image
+            style={{ alignSelf: "center" }}
+            source={require("../../img/email-checked-icon.png")}
+          />
+          <View spacer={true} extralarge={true} />
+          <H2 style={{ textAlign: "center" }}>
+            {I18n.t("reminders.email.title")}
+          </H2>
+          <View spacer={true} />
+          <Text>
+            {I18n.t("reminders.email.modal1")}
+            <Text bold={true}>{` ${profileEmail}: `}</Text>
+            {I18n.t("reminders.email.modal2")}
+          </Text>
+          <View spacer={true} />
+          <Button
+            block={true}
+            light={true}
+            bordered={true}
+            onPress={() => {
+              // https://www.pivotaltracker.com/story/show/168247365
+              Alert.alert(I18n.t("global.notImplemented"));
+            }}
+          >
+            <Text>{I18n.t("reminders.email.button1")}</Text>
+          </Button>
+        </Content>
+
+        <FooterWithButtons
+          type={"TwoButtonsInlineThirdInverted"}
+          leftButton={{
+            block: true,
+            bordered: true,
+            onPress: () => {
+              this.setState(
+                {
+                  dispatched: false
+                },
+                this.props.navigateToEmailInsertScreen
+              );
+            },
+            title: I18n.t("reminders.email.button2")
+          }}
+          rightButton={{
+            block: true,
+            primary: true,
+            onPress: () => {
```

Comment:
```suggestion
            onPress: this.handleOkPress
```
```typescript
private handleOkPress = () => {
    this.setState(
      {
        dispatched: true
      },
      this.props.validateEmail
    );
  };
```
this is because with arrow function insides tsx it becomes hard to read